services/MerchStoreService.ts

```diff
@@ -72,11 +72,18 @@ export default class MerchStoreService {
       const currentCollection = await merchCollectionRepository.findByUuid(uuid);
       if (!currentCollection) throw new NotFoundError('Collection not found');
       let updatedCollection = await merchCollectionRepository.upsertMerchCollection(currentCollection, changes);
-      if (changes.discountPercentage) {
+      if (changes.discountPercentage !== undefined) {
```

Comment:
@shravanhariharan2 I think we talked about this at some point but we should probably move towards explicit optional/maybe types.

Reply:
What do you mean by explicitly optional/maybe? Do you mean having an `Optional<T>` type, and have `discountPercentage` be `Optional<number>`, where we can just do `if (changes.discountPercentage.isNull())` (or something similar)?

Reply:
Yes.